### PR TITLE
fixes Bug 1170604 - updated use of cursors to ensure closure

### PR DESCRIPTION
**Pull request by twobraids**
close all PG cursors after their use to prevent leaking memory

### CODE REVIEW

File: socorro/external/postgresql/crashstorage.py [hunk 1]
```diff
@@ -16,6 +16,9 @@
 )
 from socorro.external.postgresql.connection_context import ConnectionContext
 from socorro.lib.datetimeutil import uuid_to_date, JsonDTEncoder
+from socorro.database.transaction_executor import (
+    TransactionExecutorWithInfiniteBackoff
```
Comment (twobraids) [inline socorro/external/postgresql/crashstorage.py]: Hmm,  looks like this was put in the wrong file as it is unnecessary here.

**Comment by peterbe** [comment]
Who should review this?

**Comment by twobraids** [comment]
I'm going to ask @selenamarie to take a look.

File: socorro/external/postgresql/dbapi2_util.py [hunk 2]
```diff
@@ -17,46 +17,46 @@ class SQLDidNotReturnSingleRow (Exception):
...
 #------------------------------------------------------------------------------
 def single_value_sql(connection, sql, parameters=None):
-    a_cursor = connection.cursor()
-    a_cursor.execute(sql, parameters)
-    result = a_cursor.fetchall()
-    try:
-        return result[0][0]
-    except Exception, x:
-        raise SQLDidNotReturnSingleValue("%s: %s" % (str(x), sql))
+    with connection.cursor() as a_cursor:
```
Comment (selenamarie) [inline socorro/external/postgresql/dbapi2_util.py]: A+ refactoring

**Comment by selenamarie** [comment]
r+

**Pull request merged**